File: duplicate-assert/java-junit/JavaParserTest_testAppendHiddenBlockCommentNodes.java
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

public class JavaParserTest extends AbstractModuleTestSupport {

    @Test
    public void testAppendHiddenBlockCommentNodes() throws Exception {
        final DetailAST root =
            JavaParser.parseFile(new File(getPath("InputJavaParserHiddenComments.java")),
                JavaParser.Options.WITH_COMMENTS);

        final Optional<DetailAST> blockComment = TestUtil.findTokenInAstByPredicate(root,
            ast -> ast.getType() == TokenTypes.BLOCK_COMMENT_BEGIN);

        assertTrue(blockComment.isPresent(), "Block comment should be present");

        final DetailAST comment = blockComment.get();

        assertEquals(3, comment.getLineNo(), "Unexpected line number");
        assertEquals(0, comment.getColumnNo(), "Unexpected column number");
        assertEquals("/*", comment.getText(), "Unexpected comment content");

        final DetailAST commentContent = comment.getFirstChild();
        final DetailAST commentEnd = comment.getLastChild();

        assertEquals(3, commentContent.getLineNo(), "Unexpected line number");
        assertEquals(2, commentContent.getColumnNo(), "Unexpected column number");
        assertEquals(9, commentEnd.getLineNo(), "Unexpected line number");
        assertEquals(1, commentEnd.getColumnNo(), "Unexpected column number");
    }
}

File: duplicate-assert/python-pytest/JavaParserTest_testAppendHiddenBlockCommentNodes.py
def test_append_hidden_block_comment_nodes():
    root = JavaParser.parseFile(
        File(getPath("InputJavaParserHiddenComments.java")),
        JavaParser.Options.WITH_COMMENTS
    )

    block_comment = TestUtil.findTokenInAstByPredicate(
        root, lambda ast: ast.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
    )

    assert block_comment.isPresent(), "Block comment should be present"

    comment = block_comment.get()

    assert comment.getLineNo() == 3, "Unexpected line number"
    assert comment.getColumnNo() == 0, "Unexpected column number"
    assert comment.getText() == "/*", "Unexpected comment content"

    comment_content = comment.getFirstChild()
    comment_end = comment.getLastChild()

    assert comment_content.getLineNo() == 3, "Unexpected line number"
    assert comment_content.getColumnNo() == 2, "Unexpected column number"
    assert comment_end.getLineNo() == 9, "Unexpected line number"
    assert comment_end.getColumnNo() == 1, "Unexpected column number"
